docs-src/java/how-to-use-a-custom-payload-codec-in-java.md

```diff
@@ -0,0 +1,111 @@
+---
+id: how-to-use-a-custom-payload-codec-in-java
+title: How to use a custom Payload Codec in Java
+sidebar_label: Custom Payload Codec
+description: Create a custom implementation of `PayloadCodec` and use it in `CodecDataConverter` to set a custom Data Converter.
+tags:
+  - Java
+  - developer-guide
+---
+
+**Create a custom Payload Codec**
+
+Create a custom implementation of [`PayloadCodec`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/payload/codec/PayloadCodec.html) and use it in [`CodecDataConverter`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/common/converter/CodecDataConverter.html) to set a custom Data Converter.
+
+The Payload Codec does byte-to-byte conversion and must be set with a Data Converter.
+
+Define custom encryption/compression logic in your `encode` method and decryption/decompression logic in your `decode` method.
+
+The following example from the [Java encryption sample](https://github.com/temporalio/samples-java/blob/main/src/main/java/io/temporal/samples/encryptedpayloads/CryptCodec.java) shows how to implement encryption and decryption logic on your payloads in your `encode` and `decode` methods.
+
+```java
+class YourCustomPayloadCodec implements PayloadCodec {
+  //...
+
+  // See the linked sample for details on the methods called here.
+  @NotNull
+  @Override
+  public List<Payload> encode(@NotNull List<Payload> payloads) {
+    return payloads.stream().map(this::encodePayload).collect(Collectors.toList());
+  }
+
+  @NotNull
+  @Override
+  public List<Payload> decode(@NotNull List<Payload> payloads) {
+    return payloads.stream().map(this::decodePayload).collect(Collectors.toList());
+  }
+
+  private Payload encodePayload(Payload payload) {
+    String keyId = getKeyId(); // Get a fixed key Id from the getKeyId method. See the sample for details.
```

Comment:
I already wrote a comment here, that getKeyId() and getKey() should be saved in this snippet on the level of signatures and a short doc. An implementation can be omitted as not important for the purpose of the sample and there should be a note about it. Otherwise, it's just an incorrect partial code.

Reply:
yeah. The idea here was to just show specific implementation example and direct the reader to the sample. But it makes sense to add them in, so updated!